lib/vsc/filesystem/lustre.py

```diff
@@ -0,0 +1,613 @@
+                    % (pjid, fsname, filesets[pjid]), LustreOperationError)
+
+            # create the fileset: dir and project
+            self.make_dir(fsetpath)
+            self._set_new_project_id(fsetpath, pjid)
+            # set inode quota
+            self._set_quota(who=pjid, obj=fsetpath, typ='project', inode_soft=inodes_max, inode_hard=inodes_max)
+            self.log.info("Created new fileset %s at %s with id %s", fileset_name, fsetpath, pjid)
+            self.set_fs_update(fsname)
+
+        else:
+        # bail if there is a fileset with the same name
+            self.log.raiseException(("Found existing fileset %s with the same name at %s ") %
+                                    (fileset_name, fsinfo['path']), LustreOperationError)
+
+
+
+    def set_user_quota(self, soft, user, obj=None, hard=None, inode_soft=None, inode_hard=None):
+        """Set quota for a user.
+
+        @type soft: integer representing the soft limit expressed in bytes
+        @type user: string identifying the user
+        @type obj: the path
+        @type hard: integer representing the hard limit expressed in bytes. If None, then 1.05 * soft.
+        @type inode_soft: integer representing the soft files limit
+        @type inode_soft: integer representing the hard files quota
+        """
+        self._set_quota(who=user, obj=obj, typ='user', soft=soft, hard=hard,
+                inode_soft=inode_soft, inode_hard=inode_hard)
+
+    def set_group_quota(self, soft, group, obj=None, hard=None, inode_soft=None, inode_hard=None):
+        """Set quota for a group on a given object (e.g., a path in the filesystem, which may correpond to a fileset)
+
+        @type soft: integer representing the soft limit expressed in bytes
+        @type group: string identifying the group
+        @type obj: the path
+        @type hard: integer representing the hard limit expressed in bytes. If None, then 1.05 * soft.
+        @type inode_soft: integer representing the soft files limit
+        @type inode_soft: integer representing the hard files quota
+        """
+        self._set_quota(who=group, obj=obj, typ='group', soft=soft, hard=hard,
+                inode_soft=inode_soft, inode_hard=inode_hard)
+
+    def set_fileset_quota(self, soft, fileset_path, hard=None, inode_soft=None, inode_hard=None):
+        """Set quota on a fileset. This maps to projects in Lustre
+
+        @type soft: integer representing the soft limit expressed in bytes
+        @type fileset_path: the linked path to the fileset
+        @type hard: integer representing the hard limit expressed in bytes. If None, then 1.05 * soft.
+        @type inode_soft: integer representing the soft files limit
+        @type inode_soft: integer representing the hard files quota
+        """
+        # we need the corresponding project id
+        project = self.get_project_id(fileset_path)
+        if int(project) == 0:
```

Comment:
get_project_id should return an int.

Reply:
I agree that's prettier, but the gpfs list_filesets used the fileset string ids as keys of dict, so I tried to keep this that way. I could convert it back for that function of course, or we should check this is no issue in the code that calls this function?